ping_manager/ping_manager.py

```diff
@@ -0,0 +1,232 @@
+import asyncio
+import contextlib
+import typing as t
+from dataclasses import asdict, dataclass, field
+from datetime import datetime, timedelta
+
+import discord
+from discord.ext import commands
+
+from bot import ModmailBot
+from core import checks
+from core.models import PermissionLevel, getLogger
+from core.thread import Thread
+from .utils import async_tasks
+
+log = getLogger(__name__)
+
+
+@dataclass
+class PingConfig:
+    """Hold the current ping configuration."""
+
+    _id: str = field(repr=False, default="ping-delay-config")
+
+    ping_string: str = "@here"
+    wait_duration: int = 5 * 60
+    ignored_categories: list[int] = field(default_factory=list)
+
+
+@dataclass(frozen=True)
+class PingTask:
+    """Data about an individual ping later task."""
+
+    when_to_close: str  # ISO datetime stamp
+    channel_id: int
+
+
+class PingManager(commands.Cog):
+    """A plugin to manage what and when to ping in ModMail threads."""
+
+    def __init__(self, bot: ModmailBot):
+        self.bot = bot
+
+        self.config: t.Optional[PingConfig] = None
+        self.ping_tasks: list[PingTask] = None
+        self.db = bot.plugin_db.get_partition(self)
+
+        self.init_task = async_tasks.create_task(self.init_plugin())
```

Comment:
FWIW, I realised the reason I got away with this here is because plugins are loaded after the main bot is already up and running, so there's always a running event loop. Specifying the bot loop explicitly is still better here though.